modules/calib3d/src/stereosgbm.cpp

```diff
@@ -114,8 +114,8 @@ static void calcPixelCostBT( const Mat& img1, const Mat& img2, int y,
                             int tabOfs, int )
 {
     int x, c, width = img1.cols, cn = img1.channels();
-    int minX1 = max(-maxD, 0), maxX1 = width + min(minD, 0);
-    int minX2 = max(minX1 - maxD, 0), maxX2 = min(maxX1 - minD, width);
+    int minX1 = max(-maxD, 0), maxX1 = width + min(-minD, 0);
+    int minX2 = max(minX1 + minD, 0), maxX2 = min(maxX1 + maxD, width);
```

Comment:
minX1, maxX2 looks for me as a set of img1 pixels that could have pair with suitable disparity. In this case we have to had a shorter search window in case of negative minD.
Also if we have [-img.width;img.width) disparity interval we get zero search window in case of original code which looks strange for such a wide disparity range